src/web/js/google-apis/api-wrapper.js

```diff
@@ -0,0 +1,358 @@
+// NOTE: Each of the following should be bound in the global scope:
+//   - `gapi'     : Google API Javascript Client
+//   - `clientId' : Google Client ID
+//   - `apiKey'   : Google API Key
+//   - `Q'        : Q Promise Framework
+
+/**
+ * The most recently authenticated version of the
+ * wrapped Google API
+ */
+var gwrap = {
+  // Initialize to a dummy method which loads the wrapper
+  load: function(params) {
+    if (!params || !params.reauth || (params.reauth.immediate === undefined)) {
+      throw new Error("Google API Wrapper not yet initialized");
+    }
+    var ret = Q.defer();
+    loadAPIWrapper(params.reauth.immediate)
+      .then(function(gw) {
+        // Shallow copy
+        var copy = $.extend({}, params);
+        delete copy.reauth;
+        gw.load(copy)
+          .then(function(loaded) {
+            ret.resolve(loaded);
+          });
+      });
+    return ret.promise;
+  },
+  request: function() {
+    throw new Error("Google API Wrapper not yet initialized");
+  }
+};
+
+// GAPI Client APIs whose methods have been wrapped with calls to gQ
+// (This is kept in the global state to avoid unneeded reloading)
+var _GWRAP_APIS = {};
+
+/**
+ * Base module for interfacing with Google APIs.
+ * Note that any APIs will only be authenticated
+ * with the OAuth scopes found in /src/google-auth.js.
+ * @param {boolean} immediate - See the description of `immediate`
```

Comment:
Is this just structured docs, or are you using something like closure compiler to check?  Cool if you are, we could work that into the build workflow.